### ISSUE TITLE
Support resource names containing a slash

### ISSUE THREAD

**Issue by anuragagarwal561994** [marmelab/admin-on-rest#1291]
**What you were expecting:**
I have a resource `user/comments` and `posts`. In the comment list I have a ReferenceField with refers to which post a comment belongs. So I have it something like:
```jsx
<ReferenceField
  label="Post"
  source="post_id"
  reference="posts"
>
  <TextField source="name" />
</ReferenceField>
```
So the URL of the link I expect it to be `/posts/:id`.

**What happened instead:**
The actual url generated is: `user/posts/:id` which is not expected.

**Environment**

* Admin-on-rest version: 1.3.2
* Last version that did not exhibit the issue (if applicable):
* React version: 15.6.2
* Browser: Any
* Stack trace (in case of a JS error):

**Comment by djhi** [comment]
If you are able to illustrate the bug or feature request with an example, please provide a sample application via one of the following means:

* CodeSandbox (https://codesandbox.io/s/ElxGNxBY0)
* A sample application via GitHub

**Comment by anuragagarwal561994** [comment]
Sample Application:
https://codesandbox.io/s/q17ymjwo9

Changed files are:

1. index.js - changed `comments` to `user/comments`.
2. data.js - changed `comments` to `user/comments`.

Notice that on url `https://q17ymjwo9.codesandbox.io/#/user/comments` the reference links to post has also been relative to `user` that is `user/posts/:id` and thus it will lead to blank page.

I could resolve the issue by changing `basePath` to `/` but is it the expected behaviour. Since I am referencing the resource name shouldn't it take it's url.

A more clear way of defining a `Resource` can be specifying both `name` and `path` as an attribute. This will allow to refer to the resource through out AOR via just the name and single point for changing the URL for the resource if the need arises in future. For example: if I am having my resources served on v3/posts why do I need to reference it every time as v3/posts, tomorrow if it changes to v4 a lot of changes will be required in the application.

**Comment by fzaninotto** [comment]
The problem is that resource names containing a slash are not supported (we never tested for that). So I'm qualifying your issue as an enhancement, and renaming it for clarity.

**Comment by anuragagarwal561994** [comment]
I think it is even okay to have resource name not support slash, providing resource url along with resource name would be more clear.

**Comment by fzaninotto** [comment]
We won't allow to provide url along with the resource name: it's the job of the restClient.

As for supporting resource names with slashes, it requires a significant refactor (changing the `basePath` to `rootPath` everywhere in the code), and is not backwards compatible. To be considered for the future.

**Comment by fzaninotto** [comment]
Fixed by #1503
